### PR TITLE
feat(tabs-extended-media): introduce tabs extended with media component

### PR DESCRIPTION
**Pull request by iwhitcomb**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5305
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5422

### Changelog

**New**

- {{new thing}}

* Tabs extended w/ Media

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

File: packages/web-components/src/components/tabs-extended/__stories__/README.stories.mdx [hunk 1]
```diff
@@ -20,6 +20,7 @@ Here's a quick example to get you started.
 
 ```javascript
 import '@carbon/ibmdotcom-web-components/es/components/tabs-extended/tabs-extended.js';
+import '@carbon/ibmdotcom-web-components/es/components/tabs-extended/tabs-extended-media.js';
```
Comment (jeffchew) [inline packages/web-components/src/components/tabs-extended/__stories__/README.stories.mdx:23]: I'm wondering if there should be two separate sections in the documentation for just `dds-tabs-extended` and for `dds-tabs-extended-media`. If an adopter only wants `dds-tabs-extended`, they won't have any need for the extended with media version so wouldn't be something they need to import.

File: packages/web-components/src/components/tabs-extended/__stories__/README.stories.mdx [hunk 2]
```diff
@@ -44,6 +45,48 @@ import '@carbon/ibmdotcom-web-components/es/components/tabs-extended/tabs-extend
...
+  </dds-tab>
+  <dds-tab label="Fourth tab">
+    <p>Content for fourth tab goes here.</p>
+  </dds-tab>
+  <dds-tab label="Fifth tab" disabled="true">
+    <p>Content for fifth tab goes here.</p>
+  </dds-tab>
+</dds-tabs-extended-media>
+```
+
```
Comment (jeffchew) [inline packages/web-components/src/components/tabs-extended/__stories__/README.stories.mdx:89]: Can't target below this, but a couple of notes:

1. Need to add `dds-tabs-extended-media` to stable selectors
2. Just doublechecking if the `markdownContents` is needed here?

File: packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  :host(#{$dds-prefix}-tabs-extended-media) {
+    margin: 0;
+    grid-column: 1 / span 16;
+    @include carbon--breakpoint(sm) {
+      padding-top: $carbon--spacing-07;
+      padding-bottom: $carbon--spacing-12;
+    }
+    @include carbon--breakpoint(lg) {
+      padding-top: $carbon--spacing-10;
+      padding-bottom: 10rem;
```
Comment (kennylam) [inline packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss:28]: Can use `$spacing-13` here.

**Review by kennylam — COMMENTED** [review]
Thanks for this @iwhitcomb! Just a small comment.

**Comment by shixiedesign** [comment]
Thank you Ian! Here are a few visual issues: 

1. Grid not aligned. The Content section title and tabs should all snap to grid: 
Current: 
image
Expected:
image


2. content section heading should top align with the Tabs labels - I've heard from Putra our content section might not be set up in the best way in terms of spacing and I'm interested in improving it. For now, if it's relatively easy, please make sure the headline and tab labels are top aligned.  
image

3. Please default `item horizontal with media` to media on the left

4. Please fill the other tabs with the `item horizontal with media` as well. Same placeholder content is fine. We wanna avoid the impression that mixing content type in tabs is allowed where in fact it's not. 

5. on mobile, content item horiz should have bottom spacing 48px. Currently too small. 
image

**Comment by kirbyevans** [comment]
1. Medium breakpoint: The media should have a max-width of 640px.
Current:
Pasted_Image_4_13_21__9_39_AM

Expected:
image

2. Small breakpoint: Media should span all four columns.
Current:
image

Expected:
image

File: packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss (hunk 3 above)
Comment (iwhitcomb) [reply packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss:28]: It doesn't look like `$carbon--spacing-13` exists.

**Comment by iwhitcomb** [comment]
Per my convo with @kirbyevans it sounds like we're going to leave the image max-width alone since it's primarily a result of the accordion styles coming from Carbon. Rather than trying to override this, we're going to stick with what Carbon provides.

> Ian Whitcomb:peanutbutterjelly:  9:47 AM
> I'm awaiting some feedback on how to answer your Q's about the max-width 640 on the image there.... that's an interesting problem.
> 
> Kirby Evans  10:07 AM
> ok cool cool. Thanks for the update.
> 
> Ian Whitcomb:peanutbutterjelly:  10:07 AM
> Basically, there are a couple things going on. I remember correctly, you can either enforce a 640px max-width OR have it lined up to the grid. There was that extra grid column with the content-item-horizontal-media that pretty enforced this on it's own so it was an easy answer...
> 
> Kirby Evans  10:08 AM
> Ah ok. hmm.
> 10:08
> so in other places it's used as an either or?
> 
> Ian Whitcomb:peanutbutterjelly:  10:09 AM
> In this particular case I'm nesting a content-item-horizontal-media which has it's own styling and rules, inside the tabs-extended-media which also has it's own set of styling and the question is whether or not the higher level component can/should dictate styles to the lower level item, first of all.
> 10:09
> But also the accordion inside tabs-extended-media has padding that's coming from Carbon. So should we attempt to change that. :man-shrugging::skin-tone-2:
> 10:09
> Override*
> 
> Kirby Evans  10:10 AM
> Ah. got it. Yeah I think the max-width rules would apply at the content-item-horizontal-media level.
> 
> Ian Whitcomb:peanutbutterjelly:  10:10 AM
> At that particular breakpoint, the accordion adds 25% padding on the right... it's weird.
> 
> 1 reply
> Today at 10:11 AMView thread
> 
> Kirby Evans  10:11 AM
> In terms of the carbon padding issue, not sure what to do there. Maybe we keep consistent with Carbon and disregard my feedback on that?
> 
> Ian Whitcomb:peanutbutterjelly:  10:11 AM
> That's exactly my question too.
> 10:11
> I'm just not sure how they typically address things like that.
> 
> Kirby Evans  10:12 AM
> Yeah, I lean staying consistent with Carbon for now. It's just a minor visual difference so I'd say let's stay consistent.
> 
> Ian Whitcomb:peanutbutterjelly:  10:13 AM
> ok, yeah.. I'll leave it alone. It seems like we're seeing a number of other issues that need to be addressed as well(i.e. the grid problem)
> 
> Kirby Evans  10:13 AM
> yeah, I think those are more important.

**Comment by shixiedesign** [comment]
Gotcha... thanks for that detailed explanation and love the diagram! Glad you are in convo with Kirby on this design. Lemme know whenever I should take another peak. It's really not far and looking great :)

**Comment by andysherman2121** [comment]
@shixiedesign Look's like Ian fixed a few of your comments. Could you provide a re-review? 
@kennylam is `$carbon--spacing-13` a usable token?

**Review by shixiedesign — APPROVED** [review]
Wooohooo! Thank you!

File: packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss (hunk 3 above)
Comment (jeffchew) [reply packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss:28]: strange, there should be here: https://carbondesignsystem.com/guidelines/spacing/overview/

Comment (andysherman2121) [reply packages/styles/scss/components/tabs-extended-media/_tabs-extended-media.scss:28]: @jeffchew it looks like it only goes up to `$carbon--spacing-13`. I just checked locally

**Review by jeffchew — APPROVED** [review]
Thank you so much @iwhitcomb !

**Pull request merged**